www/docs/server/procedures.md

```diff
@@ -5,336 +5,88 @@ sidebar_label: Define Procedures
 slug: /server/procedures
 ---
 
-Procedures in tRPC are very flexible primitives to create backend functions; they use a builder pattern which means you can create reusable base procedures for different parts of your backend application.
+In tRPC a Procedure is ultimately just a function which you want to expose to your client. A Procedure can be either:
+
+- a `Query` - used to fetch data, generally does not change any data
+- a `Mutation` - used to send data, often for create/update/delete purposes
+
+Procedures in tRPC are very flexible primitives to create backend functions, they use an immutable builder pattern which means you can [create reusable base procedures](#reusable-base-procedures) which share functionality among multiple procedures.
 
 :::tip
 
-- A procedure can be viewed as the equivalent of a REST-endpoint or a function.
+- A procedure can be viewed as the equivalent of a REST-endpoint, and under the hood are built on HTTP, but are more focused towards the needs of application, rather than representing "resources".
 - There's no internal difference between queries and mutations apart from semantics.
-- Defining publicProcedure is the same for queries, mutations, and subscription with the exception that subscriptions need to return an `observable` instance.
+- Defining a publicProcedure is the same for queries, mutations, and subscriptions, with the exception that subscriptions need to return an `observable` instance.
 
 :::
 
-## Example without input validation
+## Writing procedures
+
+The `t` object you create during tRPC setup returns an initial `t.procedure` which all other procedures are built on:
 
 ```ts twoslash
-// @filename: trpc.ts
 import { initTRPC } from '@trpc/server';
+import { z } from 'zod';
 
 const t = initTRPC.create();
 
-
-export const middleware = t.middleware;
 export const router = t.router;
 export const publicProcedure = t.procedure;
 
-// @filename: _app.ts
-// @noErrors
-// ---cut---
-import { router, publicProcedure } from './trpc';
-import { z } from 'zod';
-
 const appRouter = router({
-  // Create publicProcedure at path 'hello'
   hello: publicProcedure.query(() => {
     return {
-      greeting: 'hello world',
+      message: 'hello world',
+    };
+  }),
+  goodbye: publicProcedure.mutation(() => {
```

Comment:
i know this is "valid" tRPC usage, but do we want to show an example of a mutation being used "as a query"?

Reply:
You're right, best to set readers up for success